diff --git a/numpy/linalg/setup.py b/numpy/linalg/setup.py
index e357e29..6ef40c0 100644
--- a/numpy/linalg/setup.py
+++ b/numpy/linalg/setup.py
@@ -73,6 +73,7 @@ def configuration(parent_package='', top_path=None):
         sources=['lapack_litemodule.c', get_lapack_lite_sources],
         depends=['lapack_lite/f2c.h'],
         extra_info=lapack_info,
+        libraries=['m']
     )
 
     # umath_linalg module
@@ -81,7 +82,7 @@ def configuration(parent_package='', top_path=None):
         sources=['umath_linalg.cpp', get_lapack_lite_sources],
         depends=['lapack_lite/f2c.h'],
         extra_info=lapack_info,
-        libraries=['npymath'],
+        libraries=['npymath', 'm'],
     )
     config.add_data_files('*.pyi')
     return config