diff --git a/numpy/distutils/unixccompiler.py b/numpy/distutils/unixccompiler.py
index 4884960..d46f23e 100644
--- a/numpy/distutils/unixccompiler.py
+++ b/numpy/distutils/unixccompiler.py
@@ -124,6 +124,7 @@ def UnixCCompiler_create_static_lib(self, objects, output_libname,
         # platform intelligence here to skip ranlib if it's not
         # needed -- or maybe Python's configure script took care of
         # it for us, hence the check for leading colon.
+        self.ranlib = [os.environ.get('RANLIB')]
         if self.ranlib:
             display = '%s:@ %s' % (os.path.basename(self.ranlib[0]),
                                    output_filename)