diff --git a/numpy/distutils/system_info.py b/numpy/distutils/system_info.py
index feb28f6..c4eb1e1 100644
--- a/numpy/distutils/system_info.py
+++ b/numpy/distutils/system_info.py
@@ -371,10 +371,10 @@ if os.path.join(sys.prefix, 'lib') not in default_lib_dirs:
     default_include_dirs.append(os.path.join(sys.prefix, 'include'))
     default_src_dirs.append(os.path.join(sys.prefix, 'src'))
 
-default_lib_dirs = [_m for _m in default_lib_dirs if os.path.isdir(_m)]
-default_runtime_dirs = [_m for _m in default_runtime_dirs if os.path.isdir(_m)]
-default_include_dirs = [_m for _m in default_include_dirs if os.path.isdir(_m)]
-default_src_dirs = [_m for _m in default_src_dirs if os.path.isdir(_m)]
+default_lib_dirs = [] #[_m for _m in default_lib_dirs if os.path.isdir(_m)]
+default_runtime_dirs =[] # [_m for _m in default_runtime_dirs if os.path.isdir(_m)]
+default_include_dirs =[] # [_m for _m in default_include_dirs if os.path.isdir(_m)]
+default_src_dirs =[] # [_m for _m in default_src_dirs if os.path.isdir(_m)]
 
 so_ext = get_shared_lib_extension()
 
@@ -532,8 +532,8 @@ def get_info(name, notfound_action=0):
           'blis': blis_info,                  # use blas_opt instead
           'lapack_mkl': lapack_mkl_info,      # use lapack_opt instead
           'blas_mkl': blas_mkl_info,          # use blas_opt instead
-          'lapack_ssl2': lapack_ssl2_info,      
-          'blas_ssl2': blas_ssl2_info,          
+          'lapack_ssl2': lapack_ssl2_info,
+          'blas_ssl2': blas_ssl2_info,
           'accelerate': accelerate_info,      # use blas_opt instead
           'accelerate_lapack': accelerate_lapack_info,
           'openblas64_': openblas64__info,
@@ -939,7 +939,7 @@ class system_info:
         path = self.get_paths(self.section, key)
         if path == ['']:
             path = []
-        return path
+        return []
 
     def get_include_dirs(self, key='include_dirs'):
         return self.get_paths(self.section, key)
@@ -1164,7 +1164,7 @@ class fftw3_info(fftw_info):
                     'macros':[('SCIPY_FFTW3_H', None)]},
                   ]
 
-    
+
 class fftw3_armpl_info(fftw_info):
     section = 'fftw3'
     dir_env_var = 'ARMPL_DIR'
@@ -1855,7 +1855,7 @@ class lapack_opt_info(system_info):
     lapack_order = ['armpl', 'mkl', 'ssl2', 'openblas', 'flame',
                     'accelerate', 'atlas', 'lapack']
     order_env_var_name = 'NPY_LAPACK_ORDER'
-    
+
     def _calc_info_armpl(self):
         info = get_info('lapack_armpl')
         if info:
@@ -2049,7 +2049,7 @@ class blas_opt_info(system_info):
     blas_order = ['armpl', 'mkl', 'ssl2', 'blis', 'openblas',
                   'accelerate', 'atlas', 'blas']
     order_env_var_name = 'NPY_BLAS_ORDER'
-    
+
     def _calc_info_armpl(self):
         info = get_info('blas_armpl')
         if info:
